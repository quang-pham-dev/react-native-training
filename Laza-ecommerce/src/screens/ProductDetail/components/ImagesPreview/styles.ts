import { StyleSheet } from 'react-native';

// Themes
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  productImageReview: {
    flexDirection: 'row',
    marginVertical: Metrics.margin.xm
  },

  productImageReviewWrapper: {
    width: 77,
    height: 77,
    marginRight: 9
  },

  imagePreview: {
    width: '100%',
    height: '100%'
  }
});

export default styles;
