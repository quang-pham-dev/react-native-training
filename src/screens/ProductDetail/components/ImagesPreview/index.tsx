import React, {memo} from 'react'

// LIBS
import isEqual from 'react-fast-compare'
import {s} from 'react-native-size-matters/extend'

// Styles
import FlexStyled from '@components/Flex/Flex.styles'
import {Image} from '@components/Image/Image.styles'
import ViewStyled from '@components/View/View.styles'

// Types
import {ImageReviewerProps} from '@model-types'

type ImagesPreviewProps = {
  imagesPreview?: ImageReviewerProps[]
}
export const ImagesPreview = ({imagesPreview}: ImagesPreviewProps) => (
  <ViewStyled.ScrollViewWrapper
    horizontal
    showsHorizontalScrollIndicator={false}>
    {!!imagesPreview &&
      imagesPreview?.map(item => (
        <FlexStyled.Default key={item.id}>
          <Image.Normal
            width={s(77)}
            height={s(77)}
            mRight={s(10)}
            source={{uri: item.image}}
            resizeMode="cover"
          />
        </FlexStyled.Default>
      ))}
  </ViewStyled.ScrollViewWrapper>
)

export default memo(ImagesPreview, isEqual)
