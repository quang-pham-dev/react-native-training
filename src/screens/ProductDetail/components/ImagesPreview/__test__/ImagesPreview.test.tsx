import React from 'react'

// LIBS
import renderer from 'react-test-renderer'

// Components
import ImagesPreview from '@screens/ProductDetail/components/ImagesPreview'

// Mock data
import {product} from '@constants'

const props = {
  imagesPreview: product.imagesPreview,
}

describe('Product detail Images review', () => {
  const tree = renderer.create(<ImagesPreview {...props} />)

  test('should render correctly', () => {
    expect(tree).toMatchSnapshot()
  })
})
