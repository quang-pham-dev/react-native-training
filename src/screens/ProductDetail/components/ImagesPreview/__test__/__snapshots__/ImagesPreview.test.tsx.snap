// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product detail Images review should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    Array [
      Object {},
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Image
        height={77}
        mRight={10}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1618212158973-5e36714e374b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=774&q=80",
          }
        }
        style={
          Array [
            Object {
              "height": 77,
              "marginRight": 10,
              "resizeMode": "cover",
              "width": 77,
            },
          ]
        }
        width={77}
      />
    </View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Image
        height={77}
        mRight={10}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1516826957135-700dedea698c?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=774&q=80",
          }
        }
        style={
          Array [
            Object {
              "height": 77,
              "marginRight": 10,
              "resizeMode": "cover",
              "width": 77,
            },
          ]
        }
        width={77}
      />
    </View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Image
        height={77}
        mRight={10}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1492288991661-058aa541ff43?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=774&q=80",
          }
        }
        style={
          Array [
            Object {
              "height": 77,
              "marginRight": 10,
              "resizeMode": "cover",
              "width": 77,
            },
          ]
        }
        width={77}
      />
    </View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Image
        height={77}
        mRight={10}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1534030347209-467a5b0ad3e6?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=774&q=80",
          }
        }
        style={
          Array [
            Object {
              "height": 77,
              "marginRight": 10,
              "resizeMode": "cover",
              "width": 77,
            },
          ]
        }
        width={77}
      />
    </View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Image
        height={77}
        mRight={10}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1612462768418-d844d50931d2?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=928",
          }
        }
        style={
          Array [
            Object {
              "height": 77,
              "marginRight": 10,
              "resizeMode": "cover",
              "width": 77,
            },
          ]
        }
        width={77}
      />
    </View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Image
        height={77}
        mRight={10}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1613418035472-424391ded253?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=898&q=80",
          }
        }
        style={
          Array [
            Object {
              "height": 77,
              "marginRight": 10,
              "resizeMode": "cover",
              "width": 77,
            },
          ]
        }
        width={77}
      />
    </View>
  </View>
</RCTScrollView>
`;
